tests/test_middlewares.py

```diff
@@ -79,7 +79,46 @@ def test_hs_middlewares(hs_downloader_middleware, hs_spider_middleware):
     assert request_2.meta[HS_PARENT_ID_KEY] == 0
 
 
+@pytest.mark.skipif(sys.version_info < (3, 7), reason="requires python3.7")
+def test_hs_middlewares_dummy_response(hs_downloader_middleware, hs_spider_middleware):
+    from dataclasses import dataclass
+
+    @dataclass
+    class DummyResponse:
+        url: str
```

Comment:
Minor:

Just so that we're testing as close to scrapy-poet's `DummyResponse`, what do you think about copying the 3 lines of code from https://github.com/scrapinghub/scrapy-poet/blob/957dc34808e46059a07dc69428d5d4dca6c71ecf/scrapy_poet/api.py#L10-L31 ?

Reply:
Actually, not much code there - will copy and add, thank you @BurnzZ

Reply:
we might not need all of this code even in scrapy-poet; see https://github.com/scrapinghub/scrapy-poet/issues/99

Reply:
@kmike @BurnzZ I've changed this to the code from scrapy-poet - np revert it. Is it ok to back this to 
```
 @dataclass
    class DummyResponse:
        url: str
``` 
? I think at the moment it will be enough to test the current fix in this repo and not be tied to changes in scrapy-poet.

Reply:
so, I probably wouldn't copy the __init__ method

Reply:
Anyways, it seems it doesn't matter much; no pushback at all on merging as-is.